### PR TITLE
ImplementationChecker updates

### PR DESCRIPTION
**Pull request by dericed**
better testing and reporting of supported/unsupported formats, indentation and cleanup

### CODE REVIEW

File: MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml [hunk 1]
```diff
@@ -1,103 +1,122 @@
...
+    <reference_file>really_invalid_EBML_ElementDataSize_in_EBMLHeader.mkv</reference_file>
   </rule>
   <rule icid="EBML-VINT-EFF">
     <name>EBML vint efficiency</name>
     <authority>EBML Specification</authority>
     <target_format>EBML</target_format>
     <target_format_part>EBML Structure</target_format_part>
-    <citation>http://matroska.org/technical/specs/rfc/index.html</citation>
+    <citation></citation>
     <definition>Section 2.2 "IDs are always encoded in their shortest form, e.g. 1 is always encoded as 0x81 and never as 0x4001." The bits following the Element ID's Length Descriptor are not more than (8 - ${bit-length-of-length-descriptor}) successive 0 bits,  i.e. vint is expressed as efficiently as feasible.</definition>
```
Comment (retokromer) [inline MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml]: double space after comma: successive 0 bits,  i.e. vint

File: MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml [hunk 2]
```diff
@@ -106,29 +125,159 @@
     <target_format_part></target_format_part>
     <citation></citation>
     <definition>Ensure MKV Element ID is registered in specdata.xml (as of Dec. 13, 2014 this is 224 registered Element IDs)</definition>
```
Comment (retokromer) [inline MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml]: I prefer: as of 2014-12-13

File: MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml [hunk 3]
```diff
@@ -106,29 +125,159 @@
...
+    <note>deprecated by EBML-ELEMENT-VALID-PARENT</note>
   </rule>
   <rule icid="EBML-ELEM-SIZE-7F">
     <name>Element Size 0x7F Reservation</name>
     <authority>EBML Specification</authority>
     <target_format>EBML</target_format>
     <target_format_part>EBML Element Size</target_format_part>
-    <citation>http://matroska.org/technical/specs/rfc/index.html</citation>
+    <citation></citation>
     <definition>"Note that the shortest encoding form for 127 is 0x407f since 0x7f is reserved." If Element Size is set to 0x11111111 but element size is actually 127 bytes provide a warning.</definition>
```
Comment (retokromer) [inline MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml]: For readability, either:

- add a comma after "127 bytes"
- put "Provide a warning" before "if Element Size"

File: MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml [hunk 4]
```diff
@@ -106,29 +125,159 @@
...
+    <reference_file>element_outside_of_size_range.mkv</reference_file>
+    <note>tbd, cellar, waiting on size attribute definitions</note>
+  </rule>
+  <rule icid="EBML-NON-ASCII-IN-STRING">
+    <name>Non-Ascii Data in String</name>
+    <authority>EBML Specification</authority>
+    <target_format>EBML</target_format>
+    <target_format_part>EBML Element Size</target_format_part>
+    <citation></citation>
+    <definition>The string element is limited to certain byte ranges of ascii plus a trailing optional null byte.</definition>
```
Comment (retokromer) [inline MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml]: ASCII seems to me much more common writing than ascii. Ascii as compromise?

File: MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml [hunk 5]
```diff
@@ -106,29 +125,159 @@
...
+    <note></note>
+  </rule>
+  <rule icid="EBML-CRC-LENGTH">
+    <name>EBML CRC Element must use a valid length</name>
+    <authority>EBML Specification</authority>
+    <target_format>EBML</target_format>
+    <target_format_part>EBML Element Size</target_format_part>
+    <citation></citation>
+    <definition></definition>
+    <note>to be deprecated by EBML-WITHIN-SIZE-LIMIT with size attribute update per celler</note>
```
Comment (retokromer) [inline MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml]: cellar, CELLAR? (also below)

File: MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml (hunk 4 above)
Comment (ablwr) [inline MetadataDevelopment/ImplementationChecks/EBMLRegistry.xml]: Also in favor of ASCII full caps!

**Comment by ablwr** [comment]
In support of all of Reto's proposed changes.

**Comment by dericed** [comment]
thanks @retokromer and @ablwr all comments were acted upon, squashed, and force pushed.

**Pull request merged**